### PR TITLE
Resolves MedicalScannerVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Replaces the MedicalScannerVisualizer with a GenericVisualizerComponent
Makes it possible to insert mobs without a MobStateComponent into scanners so the "Yellow" (not alive, not dead, not critical) state is actually possible to reach.

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/MedicalScanner/MedicalScannerVisualizerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace Content.Client.MedicalScanner;
+
+[RegisterComponent]
+[Access(typeof(MedicalScannerVisualizerSystem))]
+public sealed class MedicalScannerVisualizerComponent : Component
```
Comment (AJCM-git) [inline Content.Client/MedicalScanner/MedicalScannerVisualizerComponent.cs:5]: move to MedicalScannerComponent

File: Content.Client/MedicalScanner/MedicalScannerVisualizerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace Content.Client.MedicalScanner;
+
+public sealed class MedicalScannerVisualizerSystem : VisualizerSystem<MedicalScannerVisualizerComponent>
+{
+    protected override void OnAppearanceChange(EntityUid uid, MedicalScannerVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+        if(!AppearanceSystem.TryGetData(uid, MedicalScannerVisuals.Status, out MedicalScannerStatus status, args.Component))
```
Comment (AJCM-git) [inline Content.Client/MedicalScanner/MedicalScannerVisualizerSystem.cs:13]: generic

File: Content.Client/MedicalScanner/MedicalScannerVisualizerComponent.cs (hunk 1 above)
Comment (AJCM-git) [inline Content.Client/MedicalScanner/MedicalScannerVisualizerComponent.cs:5]: move to MedicalScannerComponent

File: Content.Client/MedicalScanner/MedicalScannerVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    private string StatusToMachineStateId(MedicalScannerStatus status)
+    {
+        switch (status)
+        {
+            case Off: return "closed";
+            case Open: return "open";
+            case Red: return "occupied";
+            case Death: return "occupied";
+            case Green: return "occupied";
+            case Yellow: return "occupied";
```
Comment (AJCM-git) [inline Content.Client/MedicalScanner/MedicalScannerVisualizerSystem.cs:29]: unhardcode

File: Content.Client/MedicalScanner/MedicalScannerVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    private string StatusToTerminalStateId(MedicalScannerStatus status)
+    {
+        switch (status)
+        {
+            case Off: return "off_unlit";
+            case Open: return "idle_unlit";
+            case Red: return "red_unlit";
+            case Death: return "off_unlit";
+            case Green: return "idle_unlit";
+            case Yellow: return "maint_unlit";
```
Comment (AJCM-git) [inline Content.Client/MedicalScanner/MedicalScannerVisualizerSystem.cs:44]: same

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Client/MedicalScanner/MedicalScannerComponent.cs [hunk 5]
```diff
@@ -2,15 +2,44 @@
...
+        [MedicalScannerStatus.Off] = "off_unlit",
+        [MedicalScannerStatus.Open] = "idle_unlit",
+        [MedicalScannerStatus.Red] = "red_unlit",
+        [MedicalScannerStatus.Death] = "off_unlit",
+        [MedicalScannerStatus.Green] = "idle_unlit",
+        [MedicalScannerStatus.Yellow] = "maint_unlit",
+    };
+    #endregion Appearance
+
+    // TODO: ECS DragDrop
```
Comment (AJCM-git) [inline Content.Client/MedicalScanner/MedicalScannerComponent.cs:40]: merge fail?

File: Content.Client/MedicalScanner/MedicalScannerComponent.cs [hunk 6]
```diff
@@ -2,15 +2,44 @@
 using Content.Shared.MedicalScanner;
 using Robust.Shared.GameObjects;
 
-namespace Content.Client.MedicalScanner
+namespace Content.Client.MedicalScanner;
+
+[RegisterComponent]
+[ComponentReference(typeof(SharedMedicalScannerComponent))]
+public sealed class MedicalScannerComponent : SharedMedicalScannerComponent
```
Comment (AJCM-git) [inline Content.Client/MedicalScanner/MedicalScannerComponent.cs:9]: access

File: Content.Client/MedicalScanner/MedicalScannerComponent.cs [hunk 7]
```diff
@@ -2,15 +2,44 @@
...
+        [MedicalScannerStatus.Red] = "occupied",
+        [MedicalScannerStatus.Death] = "occupied",
+        [MedicalScannerStatus.Green] = "occupied",
+        [MedicalScannerStatus.Yellow] = "occupied",
+    };
+
+    /// <summary>
+    /// A map of the machine terminal sprite states indexed by which machine states they correspond to.
+    /// </summary>
+    public readonly Dictionary<MedicalScannerStatus, string> TerminalStateStatusMap = new()
```
Comment (AJCM-git) [inline Content.Client/MedicalScanner/MedicalScannerComponent.cs:29]: VV, datafield

File: Content.Client/MedicalScanner/MedicalScannerSystem.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Content.Client.MedicalScanner;
+
+public sealed class MedicalScannerSystem : VisualizerSystem<MedicalScannerComponent>
+{
+    protected override void OnAppearanceChange(EntityUid uid, MedicalScannerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+        if(!AppearanceSystem.TryGetData<MedicalScannerStatus>(uid, MedicalScannerVisuals.Status, out var status, args.Component))
```
Comment (AJCM-git) [inline Content.Client/MedicalScanner/MedicalScannerSystem.cs:12]: newline

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Comment by DrSmugleaf** [comment]
Reopen when you come back to this.

File: Content.Client/MedicalScanner/MedicalScannerComponent.cs (hunk 5 above)
Comment (TemporalOroboros) [reply Content.Client/MedicalScanner/MedicalScannerComponent.cs:40]: Should be fixed now.

File: Content.Client/MedicalScanner/MedicalScannerComponent.cs (hunk 6 above)
Comment (TemporalOroboros) [reply Content.Client/MedicalScanner/MedicalScannerComponent.cs:9]: Done.

File: Content.Client/MedicalScanner/MedicalScannerComponent.cs (hunk 7 above)
Comment (TemporalOroboros) [reply Content.Client/MedicalScanner/MedicalScannerComponent.cs:29]: Done.

**Comment by TemporalOroboros** [comment]
@DrSmugleaf 
Commits tested and pushed, reopen at your leisure.

File: Content.Client/MedicalScanner/MedicalScannerSystem.cs (hunk 8 above)
Comment (TemporalOroboros) [reply Content.Client/MedicalScanner/MedicalScannerSystem.cs:12]: Done.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**